pkg/storaged/content-views.jsx

```diff
@@ -883,12 +946,126 @@
                                                  */
             ];
 
+            const layouts = [
+                {
+                    value: "linear",
+                    title: _("Linear"),
+                },
+                {
+                    value: "linear_with_pvs",
+                    title: _("Linear on selected physical volumes"),
+                },
+                {
+                    value: "raid0",
+                    title: _("Striped (RAID 0)"),
+                },
+                {
+                    value: "raid1",
+                    title: _("Mirrored (RAID 1)"),
+                },
+                {
+                    value: "raid10",
+                    title: _("Striped and mirrored (RAID 10)"),
+                },
+                {
+                    value: "raid5",
+                    title: _("Distributed parity (RAID 5)"),
+                },
+                {
+                    value: "raid6",
+                    title: _("Double distributed parity (RAID 6)"),
+                }
+            ];
+
             const vdo_package = client.get_config("vdo_package", null);
             const need_vdo_install = vdo_package && !(client.features.lvm_create_vdo || client.features.legacy_vdo);
 
             if (client.features.lvm_create_vdo || client.features.legacy_vdo || vdo_package)
                 purposes.push({ value: "vdo", title: _("VDO filesystem volume (compression/deduplication)") });
 
+            const pvs_as_spaces = client.vgroups_pvols[vgroup.path].filter(pvol => pvol.FreeSize > 0).map(pvol => {
+                const block = client.blocks[pvol.path];
+                return { type: 'block', block, size: pvol.FreeSize, desc: "", pvol };
+            });
+
+            function validate_pvs(val, vals) {
+                const { layout, pvs } = vals;
+
+                if (layout == "raid0") {
+                    if (pvs.length < 2)
+                        return _("At least two physical volumes must be selected.");
+                } else if (layout == "raid1") {
+                    if (pvs.length < 2)
+                        return _("At least two physical volumes must be selected.");
+                } else if (layout == "raid10") {
+                    if (pvs.length < 4)
+                        return _("At least four physical volumes must be selected.");
+                    if (pvs.length % 2 != 0)
+                        return _("An even number of physical volumes must be selected.");
+                } else if (layout == "raid4" || layout == "raid5") {
+                    if (pvs.length < 3)
+                        return _("At least three physical volumes must be selected.");
+                } else if (layout == "raid6") {
+                    if (pvs.length < 5)
+                        return _("At least five physical volumes must be selected.");
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230531-081909-df2e1047-fedora-testing-daily//Coverage/lcov/github-pr.diff.gcov.html)